docs/misc/message_system.md

```diff
@@ -0,0 +1,242 @@
+
+- This file should never be changed manually.
+- This file is committed into the repository. 
+
+### Signing
+
+To ensure the authenticity of a configuration file, JSON Web Signatures are used. The configuration file is signed by a private key using elliptic curves (ES256) and the data specified in the config are used. The authenticity is verified on client side using corresponding public key.
+
+### CI job
+
+#### Validation
+
+- Validation of configuration file is performed in CI job in `validation` phase. It is used to detect possible structure and semantic errors. 
+- It can be run locally by `yarn workspace @trezor/suite-data msg-system-validate-config` script in suite-data.
+
+#### Signing
+
+- Signing of the configuration file is performed in CI job in `prebuild` phase.
+- The result is uploaded to `https://data.trezor.io/config/$environment/config.vX.json` and saved into `suite-data/files/message-system` to be bundled with application. For example, on localhost, the config is available at `http://localhost:3000/static/message-system/config.vX.jws`.
+- It can be run manually by `yarn workspace @trezor/suite-data msg-system-sign-config` script in suite-data. The resulting JWS is stored in `packages/suite-data/files/message-system/` in `config.vX.jws` file. 
+- Development public and private keys are baked into project structure. For production environment, these keys are replaced by CI job by production keys. This production CI job is activated on `codesign` branches.
+- Development private key can be found in `suite-data/src/message-system/constants`, the public key can be found in `suite-web` and `suite-desktop` in `next.config.js` file. *Change when tschuss-next is merged*
+
+### Versioning
+
+If changes made to the message system are incompatible with the previous version, the version number should be bumped in `messageSystemConstants.ts` in `suite` package as well as in suite-data package in `message-system/constants`. Also in `ci/packages/suite-data.yml` file.
+
+### Config Structure
+
+Structure of config, types and optionality of specific keys can be found in the schema or in generated types. Example config is commented below.
+
+```javascript
+{
+    // Version of message system implementation. Bump if new version is not backward compatible.
+    "version": 1, 
+    // Datetime in ISO8601 when was config created.
+    "timestamp": "2021-03-03T03:48:16+00:00",
+    // Version of config. New config is accepted only if sequence number is higher.
+    "sequence": 1, 
+    "actions": [
+        {
+            /*
+            - User's stack has to match one of the condition the object to show this message.
+            - The bitwise operation is AND.
+            */
+            "conditions": [ 
+                /* 
+                - The bitwise operation is AND.
```

Comment:
I think OR (union) might be more useful. Let's say I want to target Firefox 80 on Mac and Firefox 82 on Windows with a same message. If I understand it correctly, I'd have to create two separate actions if AND is used here.

Reply:
Bug in the documentation. The operation between object conditions is indeed OR.